chromium_src/tools/json_schema_compiler/feature_compiler.py

```diff
@@ -0,0 +1,218 @@
+# Copyright (c) 2023 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pylint: disable=line-too-long,too-few-public-methods,undefined-variable,protected-access
+
+import override_utils
+
+
+# echo -n mnojpmjdmbbfmejpflffifhffcmidifd | openssl sha1 | tr '[:lower:]' '[:upper:]'
+class BraveExtensions:
+    BRAVE_SHIELDS = "A321D47A2B4CA86898167A55CA8B2E02385EA7CD"  # mnojpmjdmbbfmejpflffifhffcmidifd
+    ETHEREUM_REMOTE_CLIENT = "21070F3D60711361C1210B870439BE49B5D995F4"  # odbfpeeihdkbihmopkbjmoonfanlbfcl
+    IPFS_COMPANION = "780BF954C0F7C586EA9662D4F967771F49CC2114"  # nibjojkomfdiaoajekhjakgkdhaomnch
+    IPFS_COMPANION_BETA = "FF32507DC3DB5DFFD1D6733187C84D4B74713D63"  # hjoieblefckbooibpepigmacodalfndh
+    WEB_TORRENT = "3D9518A72EB02667A773B69DBA9E72E0F4A37423"  # lgjmpdmojkpocjcopdikifhejkkjglho
+
+
+class BraveFeatureDefinitionExtender:
+    # This is executed for all important files. List all of them so we can fail
+    # in case something new appears in the repository or if an existing file is
+    # being moved somewhere.
+    KNOWN_FILES = {
+        "chrome/common/apps/platform_apps/api/_api_features.json",
+        "chrome/common/apps/platform_apps/api/_permission_features.json",
+        "chrome/common/controlled_frame/api/_api_features.json",
+        "chrome/common/extensions/api/_api_features.json",
+        "chrome/common/extensions/api/_manifest_features.json",
+        "chrome/common/extensions/api/_permission_features.json",
+        "extensions/common/api/_api_features.json",
+        "extensions/common/api/_behavior_features.json",
+        "extensions/common/api/_manifest_features.json",
+        "extensions/common/api/_permission_features.json",
+    }
+
+    # Feature definitions to be added to existing features.
+    FEATURE_DEFINITIONS_TO_ADD = {
```

Comment:
should we put these in a separate file?

Reply:
> should we put these in a separate file?

created mirrored `//brave/chromium_src/...` files to adjust extension APIs.

initially I put everything into `feature_compiler.py`, because it was unclear how `json` merge should work if we want to replace some specific API entirely instead of merging it. Although we don't have such use cases now, but I've implemented this via `replace!` prefix that can exist in overridden jsons. This way we can use mirrored jsons to store feature definition extensions properly.